src/common.ts

```diff
@@ -1,23 +1,86 @@
-import type {HtmlController} from './runtime/HtmlController';
+import {useEffect, useState} from 'react';
 
-export const GLOBAL_SYMBOL: unique symbol = Symbol.for('diplodocHtml');
+type Callback<T> = (controller: T) => void;
 
-declare global {
-    interface Window {
-        [GLOBAL_SYMBOL]: HtmlController;
-    }
+export type ScriptStore<T> = Callback<T>[] | null;
+
+export interface CreateLoadQueueArgs<T> {
+    store: ScriptStore<T>;
+    controller: T;
 }
 
-export const BLOCK_NAME = 'yfm-html';
-export const HTML_DATA_ID = 'data-diplodoc-id';
-export const HTML_DATA_KEY = 'data-diplodoc-key';
+const w = window as any;
+
+export const isBrowser = () => typeof w !== 'undefined' && typeof document !== 'undefined';
```

Comment:
In nodejs environment, runtime error will be there:
`Uncaught ReferenceError: window is not defined`

We always need to check window as follows: `typeof window !== 'undefined'`. And only after that we can use window as a global variable

Reply:
fixed